components/elm/src/biogeochem/AllocationMod.F90

```diff
@@ -672,7 +672,7 @@ subroutine Allocation1_PlantNPDemand (bounds, num_soilc, filter_soilc, num_soilp
 
          f5 = 0._r8 ! continued intializations from above
 
-         if (ivt(p) >= npcropmin) then ! skip 2 generic crops
+         if (iscft(ivt(p)) >= 1) then ! skip 2 generic crops
```

Comment:
@rgknox there are a number of flags that are currently reals but are intended as logical (usually named is<something>). In the spirit of getting this set up for a broader overhaul later on, I used iscft as an example and did the conversion throughout the code to a logical variable in the latest commit. That allows these kinds of tests to be of the form if (iscft(ivt(p))) then ..., or if (.not. iscft(ivt(p))) then ...   
That seems like a clear approach for the logical flags. The use of multiple values for a flag like woody is going to take some additional thought and work, since it is sometimes used like if (woody > 1), which does not lend itself any better to a named value replacing the 1.
Hopefully this demonstration with iscft shows an acceptable path for future development, while allowing this PR to move ahead.